### ISSUE TITLE
The prompt color has changed before I complete it

### ISSUE THREAD

**Issue by jiangtao69039** [zsh-users/zsh-syntax-highlighting#862]
I didn't press the right arrow button,but the color has changed. how can i resolve this problem?

![1](https://user-images.githubusercontent.com/20572131/156364505-d3b4018b-0595-497b-ae70-a8f346910e34.png)
![2](https://user-images.githubusercontent.com/20572131/156364513-c89b48de-6c58-416b-9aee-985de8728b30.png)

**Comment by danielshahaf** [comment]
Please post the output of `typeset -p ZSH_VERSION ZSH_PATCHLEVEL ZSH_HIGHLIGHT_REVISION ZSH_HIGHLIGHT_VERSION`.  (Do not upload a screenshot; copy-paste it as text, so we can copy-paste the values on our side.)

**Comment by Pcholken** [comment]
+1
typeset ZSH_VERSION=5.8.1
typeset ZSH_PATCHLEVEL=zsh-5.8.1-0-g1a490c7
typeset ZSH_HIGHLIGHT_REVISION=HEAD
typeset ZSH_HIGHLIGHT_VERSION=0.8.0-alpha2-dev

**Comment by jiangtao69039** [comment]
> Please post the output of `typeset -p ZSH_VERSION ZSH_PATCHLEVEL ZSH_HIGHLIGHT_REVISION ZSH_HIGHLIGHT_VERSION`. (Do not upload a screenshot; copy-paste it as text, so we can copy-paste the values on our side.)

i've resolved by update zsh-syntax-highlighting plugin.

**Comment by jiangtao69039** [comment]
> +1 typeset ZSH_VERSION=5.8.1 typeset ZSH_PATCHLEVEL=zsh-5.8.1-0-g1a490c7 typeset ZSH_HIGHLIGHT_REVISION=HEAD typeset ZSH_HIGHLIGHT_VERSION=0.8.0-alpha2-dev

i've resolved by this command:
```shell
cd ~/.oh-my-zsh/custom/plugins/zsh-syntax-highlighting
git pull
```

**Comment by danielshahaf** [comment]
I thought that might be the case :-)

Duplicate of #857.

**Comment by danielshahaf** [comment]
@jiangtao69039 For future reference, posting the same question verbatim to multiple projects in parallel is considered impolite since it duplicates the supporters' efforts.  It's best avoided, or failing that, the various tickets should link to each other to prevent duplication of effort.

**Comment by Silverbullet069** [comment]
I'm having the same problem again.

After typing the first character:
![1](https://user-images.githubusercontent.com/80348187/231940416-6aaffbdb-fc22-4fee-87e0-f86fb072ac2a.jpg)

After typing 2nd:
![2](https://user-images.githubusercontent.com/80348187/231940460-75f4b310-7329-485b-a7f1-4aa36c75a160.jpg)

Full (without spacebar):
![3](https://user-images.githubusercontent.com/80348187/231940521-31d2c109-1a09-4182-aaea-e69fee0c3fe4.jpg)

Full (with spacebar):
![4](https://user-images.githubusercontent.com/80348187/231940547-5fc0817f-b1f8-44ce-8ce0-744d56fe3470.jpg)

typeset ZSH_VERSION=5.8.1
typeset ZSH_PATCHLEVEL=ubuntu/5.8.1-1
typeset ZSH_HIGHLIGHT_REVISION=HEAD
typeset ZSH_HIGHLIGHT_VERSION=0.8.0-alpha2-dev
